website/content/docs/developers/packing/ffdpacking.md

```diff
@@ -0,0 +1,61 @@
+---
+title: The First Fit Decreasing Packing Algorithm
+---
+
+### Overview
+
+This guide provides basic steps at using and tuning the First Fit Decreasing packing algorithm in order to  utilize the resources efficiently. This packing algorithm aims at utilizing as few containers as possible, thus limiting the overall resources used. The algorithm is based on the First Fit Decreasing heuristic for the binpacking problem. To enable the First Fit Decreasing algorithm, update the `packing.yaml` file as follows:
+
+```yaml
+# packing algorithm for packing instances into containers
+heron.class.packing.algorithm:    com.twitter.heron.packing.binpacking.FirstFitDecreasingPacking
+```
+
+The algorithm accepts as input the values of the following parameters:
+
+1. Component RAM 
+2. Hint for the maximum container RAM (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_MAX_RAM_HINT`)
+3. Hint for the maximum Container CPU (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_MAX_CPU_HINT`)
+4. Hint for the maximum Container Disk (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_MAX_DISK_HINT`)
+5. Padding percentage (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_PADDING_PERCENTAGE`)
+6. Component Parallelisms
+
+Parameter 1 determines the RAM requirement of each component in the topology.
+If the requirement is not specified then a default value of 1GB is used. The First Fit Decreasing algorithm
+guarantees that the amount of memory allocated to a component is either the one specified by the user
+or the default one.
+
+The parameters 2-4 determine the maximum container size with respect to RAM, cpu cores and disk. 
+If one of these parameters is not specified by the user then
+the hint for the corresponding maximum container resource is set to the default resource requirement of 4 Heron instances. 
+
+Note that these values take into account only the resources allocated for the user's instances. Additional per container resources for system-related processes such as the stream manager can be added to the maximum container size defined above. Thus, the algorithm might eventually produce containers whose size is slighltly bigger that the boundary determined by parameters 2-4.
```

Comment:
just a nit, but please line break at 100 chars to make commenting and reading in an IDE easier.